### PR TITLE
Add support for modifying mapUsers key in aws-auth configmap

### LINKED ISSUE #876: Add `mapUsers` support to `iamidentitymapping` command
With the recent addition of the `iamidentitymapping` command I noticed that it currently only supports the `mapRoles` functionality of the `aws-auth` configmap. It would be nice to have it support the `mapUsers` functionality as well. In other words get/create/delete mappings for both IAM roles and users to Kubernetes users and groups.

### PR DESCRIPTION
**Pull request by rndstr**
Replaces #932

Fixes #876

### Checklist

- [x] Code compiles correctly (i.e `make build`)
- [x] Added tests that cover your change (if possible)
- [x] All unit tests passing (i.e. `make test`)
- [x] Added/modified documentation as required (such as the `README.md`, and `examples` directory)
- [x] Manually tested

### CODE REVIEW

**Comment by martina-if** [comment]
I still need to manually test the changes

**Comment by errordeveloper** [comment]
@rndstr @martina-if thanks for carrying this forward, I'm hoping to have a good look at the diff next week!

File: pkg/iam/mapping.go [hunk 1]
```diff
@@ -1,17 +1,117 @@
...
+	ErrNeitherUserNorRole = errors.New("arn is neither user nor role")
+
+	// ErrNoKubernetesIdentity is the error returned when an identity has neither a Kubernetes
+	// username nor a list of groups.
+	ErrNoKubernetesIdentity = errors.New("neither username nor groups is set for iam identity")
+)
+
+// Identity represents an IAM identity and its corresponding Kubernetes identity
+type Identity interface {
+	GetARN() string
```
Comment (rndstr) [inline pkg/iam/mapping.go]: Not a fan of the `Get` prefixes here but `Username` and `Groups` will conflict. I'm OK with leaving the `GetARN()` for consistency (at least within the struct) but also don't oppose removing it.

An alternative would be to unexport fields and implement `MarshalJSON()` where one uses a copy of the struct with exported fields but that's also quite verbose.

I'm fine as is.

Comment (martina-if) [reply pkg/iam/mapping.go]: Yeah, I don't like it either. What if I make the structs have different field names:

```
type RoleIdentity struct {
  RoleARN,
  {
    KubernetesUsername string `json:"username", omitempty`,
    KubernetesGroups []string `json:"groups", omitempty`,
  }
}

type Identity interface {
  ARN()
  Username()
  Groups()
}
```

Comment (rndstr) [reply pkg/iam/mapping.go]: i like it!

Comment (rndstr) [reply pkg/iam/mapping.go]: I refactored it (and rebased)

File: pkg/iam/mapping.go [hunk 2]
```diff
@@ -1,17 +1,109 @@
...
-	Username string   `json:"username"`
-	Groups   []string `json:"groups"`
+var (
+	// ErrNeitherUserNorRole is the error returned when an identity is missing both UserARN
+	// and RoleARN.
+	ErrNeitherUserNorRole = errors.New("arn is neither user nor role")
+
+	// ErrNoKubernetesIdentity is the error returned when an identity has neither a Kubernetes
+	// username nor a list of groups.
+	ErrNoKubernetesIdentity = errors.New("neither username nor groups is set for iam identity")
```
Comment (martina-if) [inline pkg/iam/mapping.go]:
```suggestion
	ErrNoKubernetesIdentity = errors.New("neither username nor group are set for iam identity")
```

File: pkg/ctl/cmdutils/iam_flags.go [hunk 3]
```diff
@@ -17,15 +16,6 @@ func AddIAMServiceAccountFilterFlags(fs *pflag.FlagSet, includeGlobs, excludeGlo
...
-	_ = fs.MarkDeprecated("role", "see --arn")
-	AddPreRunE(cmd.CobraCommand, func(cobraCmd *cobra.Command, args []string) error {
-		var err error
-		if role != "" {
-			*arn = role
-		}
-		return err
-	})
+	fs.StringVar(arn, "role", "", "")
+	_ = fs.MarkDeprecated("role", "use --arn instead")
```
Comment (martina-if) [inline pkg/ctl/cmdutils/iam_flags.go]: Don't we need to keep supporting it for a couple of versions before deprecating it completely?

**Comment by martina-if** [comment]
I rebased and ran the integration tests. I think this is ready to merge!

**Comment by neilharris123** [comment]
Is there any documentation available for this feature? I'm trying to map a user to a group using eksctl but can't find any instructions on how to utilise the mapusers feature suggested by this PR.

**Pull request merged**